Content.Client/CombatMode/ShowCombatModeIndicatorsOverlay.cs

```diff
@@ -0,0 +1,113 @@
+using Content.Shared.CCVar;
+using Content.Client.Hands.Systems;
+using Content.Shared.Weapons.Ranged.Components;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.Input;
+using Robust.Client.UserInterface;
+using Robust.Client.Player;
+using Robust.Shared.Configuration;
+using Robust.Shared.Enums;
+using Robust.Shared.Utility;
+
+namespace Content.Client.CombatMode
+{
+    public sealed class ShowCombatModeIndicatorsOverlay : Overlay
+    {
+        [Dependency] private readonly IConfigurationManager _cfg = default!;
+        [Dependency] private readonly IInputManager _inputManager = default!;
+        [Dependency] private readonly IEntityManager _entMan = default!;
+        [Dependency] private readonly IEyeManager _eye = default!;
+        private Texture? _gunSight;
+        private Texture? _meleeSight;
+        private CombatModeSystem _combatSystem;
+        private IPlayerManager _player;
+        public override OverlaySpace Space => OverlaySpace.ScreenSpace;
+        public ShowCombatModeIndicatorsOverlay(IPlayerManager playerManager, CombatModeSystem combatSys)
+        {
+            IoCManager.InjectDependencies(this);
+
+            _combatSystem = combatSys;
+            _player = playerManager;
+
+            _gunSight = GetTextureFromRsi("gunsight");
+            _meleeSight = GetTextureFromRsi("meleesight");
+        }
+
+        private Texture GetTextureFromRsi(string _spriteName)
+        {
+            var sprite = new SpriteSpecifier.Rsi(
+                new ResPath("/Textures/Interface/Misc/pointer_sights.rsi"), _spriteName);
+            return _entMan.EntitySysManager.GetEntitySystem<SpriteSystem>().Frame0(sprite);
+        }
+
+        protected override void DisposeBehavior()
+        {
+            base.DisposeBehavior();
+        }
+
+        protected override void Draw(in OverlayDrawArgs args)
+        {
+            if (!_cfg.GetCVar(CCVars.HudHeldItemShow))
+                return;
```

Comment:
The CCVar should just toggle the overlay instead of checking every frame.

Reply:
fixed